internal/github/ratelimit_test.go

```
@@ -92,3 +106,94 @@ func TestMergeRateLimit3(t *testing.T) {
+// FetchRateLimits must decode the /rate_limit payload (including
+// the unix-seconds reset) and classify HTTP failures through the
+// shared FetchError reasons.
+func TestFetchRateLimits(t *testing.T) {
+	t.Run("decodes and orders", func(t *testing.T) {
+		reset := time.Now().Add(30 * time.Minute).Unix()
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path != "/rate_limit" {
+				t.Errorf("path = %q, want /rate_limit", r.URL.Path)
+			}
+			w.Header().Set("Content-Type", "application/json")
+			w.Write([]byte(`{"resources":{` +
```

golangci-lint (2.12.2)

[error] Error return value of `w.Write` is not checked

(errcheck)